sklearn/decomposition/tests/test_pca.py

```diff
@@ -529,6 +529,26 @@ def test_pca_score3():
     assert_true(ll.argmax() == 1)
 
 
+def test_pca_score4():
```

Comment:
@lesteve Thanks. The reason for current test is that noise_variance is not correctly implemented in PCA._fit_truncated but correctly implemented in PCA._fit_full, which is the main purpose of the pull request. So we can compare between the scores calculated with different methods to construct the test.